pkg/service/roomservice.go

```diff
@@ -75,6 +81,21 @@ func (s *RoomService) DeleteRoom(ctx context.Context, req *livekit.DeleteRoomReq
 		return nil, err
 	}
 
+	// we should not return until when the room is confirmed deleted
+	err = confirmExecution(func() error {
```

Comment:
room service functions had a shortcoming that they returned too soon before the operation has actually completed on the media node. this has led to unpredictable behavior for users but also our tests.